__tests__/helpers/url.spec.js

```diff
@@ -47,6 +47,18 @@ describe('buildFetchUrl', () => {
     expect(url).toBe('http://api.domain.com/users/123/posts');
   });
 
+  it('should handle empty optional ids by removing extra forward slashes or question marks', () => {
+    const baseUrl = 'http://api.domain.com/users/:userId?/posts/:postId';
+    const url = buildFetchUrl({ url: baseUrl, id: { someId: 'huh' } });
+    expect(url).toBe('http://api.domain.com/users/posts');
+  });
+
+  it('should handle non-empty optional ids', () => {
```

Comment:
Can you add a test that has a URL with query params that use a question mark?
e.g. `http://api.domain.com/users?userid=:userId`

Reply:
This test passes:
```
    const baseUrl = 'http://api.domain.com/users/:userId';
    const url = buildFetchUrl({
      url: baseUrl,
      opts: {
        query: { userId: 'questionable?' },
      },
    });
    expect(url).toBe('http://api.domain.com/users?userId=questionable?');
```
However if I change the baseUrl to end with `/:userId?`, an extra ampersand appears:
![image](https://user-images.githubusercontent.com/967811/78714528-0f253b00-78ea-11ea-8445-ff5c7b5125bb.png)

Would it make sense to include in the readme that marking a path parameter as optional by adding the question mark only works for parameters not at the end of the url? Because for urls with the path param is at the end, it is inherently optional.

Reply:
To me it seems like there is still an underlying bug if it's giving surprising results, and I don't think adding it to the README is an effective solution.

Reply:
Agreed. I guess the `.replace(/\?\//g, '/')` addition converting `?/` matches to `/` also needs to eliminate the trailing `?` if there is one like in this case. But since there's no need to apply that regex to the query params, I'll add it to the end of `replaceUrlParams({ url: remainderUrl, id })` instead